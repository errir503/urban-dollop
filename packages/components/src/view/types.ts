/**
 * External dependencies
 */
import type { ReactNode } from 'react';

export type ViewProps = { children?: ReactNode };
