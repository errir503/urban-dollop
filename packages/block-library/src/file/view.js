/**
 * Internal dependencies
 */
import { hidePdfEmbedsOnUnsupportedBrowsers } from './utils';

document.addEventListener(
	'DOMContentLoaded',
	hidePdfEmbedsOnUnsupportedBrowsers
);
