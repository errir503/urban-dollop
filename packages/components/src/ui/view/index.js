export { default as View } from './component';
