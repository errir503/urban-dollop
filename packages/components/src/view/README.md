# View

<div class="callout callout-alert">
This feature is still experimental. “Experimental” means this is an early implementation subject to drastic and breaking changes.
</div>

`View` is a core component that renders everything in the library. It is the principle component in the entire library.

**Everything** is a `View`, and a `View` is **everything**.

## Usage

```jsx
import {
	__experimentalText as Text,
	__experimentalView as View,
} from '@wordpress/components';

function Example() {
	return (
		<View>
			<Text>Code is Poetry</Text>
		</View>
	);
}
```

## Props

### as

**Type**: `string`,`E`

Render the component as another React Component or HTML Element.

### css

**Type**: `InterpolatedCSS`

Render custom CSS using the style system.
