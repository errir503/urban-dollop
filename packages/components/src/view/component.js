/**
 * External dependencies
 */
import styled from '@emotion/styled';

/**
 * `View` is a core component that renders everything in the library.
 * It is the principle component in the entire library.
 *
 * @example
 * ```jsx
 * import { View } from `@wordpress/components`;
 *
 * function Example() {
 * 	return (
 * 		<View>
 * 			 Code is Poetry
 * 		</View>
 * 	);
 * }
 * ```
 *
 * @type {import('../ui/context').WordPressComponent<'div', { children?: import('react').ReactNode }, true>}
 */
// @ts-ignore
const View = styled.div``;
View.selector = '.components-view';
View.displayName = 'View';

export default View;
