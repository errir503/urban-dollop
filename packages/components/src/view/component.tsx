/**
 * External dependencies
 */
import styled from '@emotion/styled';
import type { RefAttributes } from 'react';

/**
 * Internal dependencies
 */
import type { WordPressComponent } from '../ui/context/wordpress-component';
import type { ViewProps } from './types';

/**
 * `View` is a core component that renders everything in the library.
 * It is the principle component in the entire library.
 *
 * ```jsx
 * import { View } from `@wordpress/components`;
 *
 * function Example() {
 * 	return (
 * 		<View>
 * 			 Code is Poetry
 * 		</View>
 * 	);
 * }
 * ```
 */
export const View: WordPressComponent<
	'div',
	ViewProps & RefAttributes< any >,
	true
> = styled.div``;

View.selector = '.components-view';
View.displayName = 'View';

export default View;
