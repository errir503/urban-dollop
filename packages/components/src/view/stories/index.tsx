/**
 * External dependencies
 */
import type { ComponentMeta, ComponentStory } from '@storybook/react';

/**
 * Internal dependencies
 */
import { View } from '..';

const meta: ComponentMeta< typeof View > = {
	component: View,
	title: 'Components (Experimental)/View',
	argTypes: {
		as: { control: { type: null } },
		children: { control: { type: 'text' } },
	},
	parameters: {
		controls: { expanded: true },
		docs: { source: { state: 'open' } },
	},
};
export default meta;

const Template: ComponentStory< typeof View > = ( args ) => {
	return <View { ...args } />;
};

export const Default: ComponentStory< typeof View > = Template.bind( {} );
Default.args = {
	children: 'An example tip',
};
