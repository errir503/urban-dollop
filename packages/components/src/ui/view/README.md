# View (Experimental)

`View` is a core component that renders everything in the library. It is the principle component in the entire library.

**Everything** is a `View`, and a `View` is **everything**.

## Usage

```jsx
import { Text, View } from '@wordpress/components/ui';

function Example() {
	return (
		<View>
			<Text>Code is Poetry</Text>
		</View>
	);
}
```

## Props

##### as

**Type**: `string`,`E`

Render the component as another React Component or HTML Element.

##### css

**Type**: `InterpolatedCSS`

Render custom CSS using the style system.
