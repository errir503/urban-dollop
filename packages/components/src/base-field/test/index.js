/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import { useBaseField } from '../index';
import { View } from '../../view';

const TestField = ( props ) => {
	return <View { ...useBaseField( props ) } />;
};

describe( 'base field', () => {
	let base;

	beforeEach( () => {
		base = render( <TestField /> ).container;
	} );

	it( 'should render correctly', () => {
		expect( base.firstChild ).toMatchSnapshot();
	} );

	describe( 'props', () => {
		it( 'should render error styles', () => {
			const { container } = render( <TestField hasError /> );
			expect( container.firstChild ).toMatchStyleDiffSnapshot(
				base.firstChild
			);
		} );

		it( 'should render inline styles', () => {
			const { container } = render( <TestField isInline /> );
			expect( container.firstChild ).toMatchStyleDiffSnapshot(
				base.firstChild
			);
		} );

		it( 'should render subtle styles', () => {
			const { container } = render( <TestField isSubtle /> );
			expect( container.firstChild ).toMatchStyleDiffSnapshot(
				base.firstChild
			);
		} );
	} );

	describe( 'useBaseField', () => {
		it( 'should pass through disabled and defaultValue props', () => {
			// wrap this in a component so that `useContext` calls don't fail inside the hook
			// assertions will still run as normal when we `render` the component :)
			const Component = () => {
				const disabled = Symbol.for( 'disabled' );
				const defaultValue = Symbol.for( 'defaultValue' );

				const result = useBaseField( { disabled, defaultValue } );

				expect( result.disabled ).toBe( disabled );
				expect( result.defaultValue ).toBe( defaultValue );

				return null;
			};

			render( <Component /> );
		} );
	} );
} );
