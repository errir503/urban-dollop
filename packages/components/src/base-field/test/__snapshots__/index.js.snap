// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base field props should render error styles 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

@@ -11,11 +11,11 @@
      "-webkit-justify-content": "space-between",
      "-webkit-transition": "border-color 100ms ease",
      "align-items": "center",
      "background": "#fff",
      "border": "1px solid",
-     "border-color": "#d94f4f",
+     "border-color": "#757575",
      "border-radius": "2px",
      "box-shadow": "transparent",
      "display": "flex",
      "flex": "1",
      "flex-direction": "row",
`;

exports[`base field props should render inline styles 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

@@ -14,18 +14,17 @@
      "background": "#fff",
      "border": "1px solid",
      "border-color": "#757575",
      "border-radius": "2px",
      "box-shadow": "transparent",
-     "display": "inline-flex",
+     "display": "flex",
      "flex": "1",
      "flex-direction": "row",
      "font-size": "13px",
      "justify-content": "space-between",
      "outline": "none",
      "padding": "0 8px",
      "position": "relative",
      "transition": "border-color 100ms ease",
-     "vertical-align": "baseline",
-     "width": "auto",
+     "width": "100%",
    },
  ]
`;

exports[`base field props should render subtle styles 1`] = `
Snapshot Diff:
- Received styles
+ Base styles

@@ -10,11 +10,10 @@
      "-webkit-flex-direction": "row",
      "-webkit-justify-content": "space-between",
      "-webkit-transition": "border-color 100ms ease",
      "align-items": "center",
      "background": "#fff",
-     "background-color": "transparent",
      "border": "1px solid",
      "border-color": "#757575",
      "border-radius": "2px",
      "box-shadow": "transparent",
      "display": "flex",
`;

exports[`base field should render correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  width: 100%;
  background: #fff;
  border-radius: 2px;
  border: 1px solid;
  border-color: #757575;
  box-shadow: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 13px;
  outline: none;
  padding: 0 8px;
  position: relative;
  -webkit-transition: border-color 100ms ease;
  transition: border-color 100ms ease;
  width: 100%;
}

.emotion-0>*+*:not( marquee ) {
  margin-left: calc(4px * 2);
}

.emotion-0>* {
  min-width: 0;
}

@media ( prefers-reduced-motion: reduce ) {
  .emotion-0 {
    transition-duration: 0ms;
  }
}

.emotion-0[disabled] {
  opacity: 0.6;
}

.emotion-0:hover {
  border-color: #757575;
}

.emotion-0:focus,
.emotion-0[data-focused='true'] {
  border-color: var( --wp-admin-theme-color, #00669b);
  box-shadow: 0 0 0 0.5px var( --wp-admin-theme-color, #00669b);
}

<div
  class="components-flex components-base-field emotion-0 emotion-1"
  data-wp-c16t="true"
  data-wp-component="BaseField"
/>
`;
